networkx/readwrite/gexf.py

```diff
@@ -193,47 +193,48 @@ class GEXF:
     }
     versions["1.2draft"] = d
 
-    types = [
-        (int, "integer"),
-        (float, "float"),
-        (float, "double"),
-        (bool, "boolean"),
-        (list, "string"),
-        (dict, "string"),
-        (int, "long"),
-        (str, "liststring"),
-        (str, "anyURI"),
-        (str, "string"),
-    ]
-
-    # These additions to types allow writing numpy types
-    try:
-        import numpy as np
-    except ImportError:
-        pass
-    else:
-        # prepend so that python types are created upon read (last entry wins)
+    def construct_types(self):
         types = [
-            (np.float64, "float"),
-            (np.float32, "float"),
-            (np.float16, "float"),
-            (np.float_, "float"),
-            (np.int_, "int"),
-            (np.int8, "int"),
-            (np.int16, "int"),
-            (np.int32, "int"),
-            (np.int64, "int"),
-            (np.uint8, "int"),
-            (np.uint16, "int"),
-            (np.uint32, "int"),
-            (np.uint64, "int"),
-            (np.int_, "int"),
-            (np.intc, "int"),
-            (np.intp, "int"),
-        ] + types
-
-    xml_type = dict(types)
-    python_type = dict(reversed(a) for a in types)
+            (int, "integer"),
+            (float, "float"),
+            (float, "double"),
+            (bool, "boolean"),
+            (list, "string"),
+            (dict, "string"),
+            (int, "long"),
+            (str, "liststring"),
+            (str, "anyURI"),
+            (str, "string"),
+        ]
+
+        # These additions to types allow writing numpy types
+        try:
+            import numpy as np
+        except ImportError:
+            pass
+        else:
+            # prepend so that python types are created upon read (last entry wins)
+            types = [
+                (np.float64, "float"),
+                (np.float32, "float"),
+                (np.float16, "float"),
+                (np.float_, "float"),
+                (np.int_, "int"),
+                (np.int8, "int"),
+                (np.int16, "int"),
+                (np.int32, "int"),
+                (np.int64, "int"),
+                (np.uint8, "int"),
+                (np.uint16, "int"),
+                (np.uint32, "int"),
+                (np.uint64, "int"),
+                (np.int_, "int"),
+                (np.intc, "int"),
+                (np.intp, "int"),
+            ] + types
+
+        self.xml_type = dict(types)
+        self.python_type = dict(reversed(a) for a in types)
```

Comment:
This code seems to rely on getting the items in `types` in a particular order. Several keys here will not be unique and you are relying on the CPython implementation to always overwrite the duplicated values. 

This also has an issue that `'int'` maps to `numpy.int64` iff numpy is present. Did you mean to use "integer" there like you did with the builtin types? I would recommend changing the first `types` variable to `builtin_types` and then have the second one be `numpy_types`. Then conditionally build `types` as `builtin_types + numpy_types` or just `builtin_types` (this has the added benefit of making the code idempotent).

Then I would rewrite the `python_type` line as

```python
self.python_type = {v: k for k, v in builtin_types}
```

which avoids any ambiguity. Biggest issue though is that "int" versus "integer" key. Not sure if you meant to do that.

Reply:
This comment should be addressed in a different PR